File: C/convert_binary_number_in_a_linked_list_to_integer.c
#include <stddef.h>
#include <stdlib.h>
#include <stdio.h>


struct ListNode
{
    int val;
    struct ListNode *next;
};

int getDecimalValue(struct ListNode *head)
{
    char result[100];
    struct ListNode *cur = head;
    int i = 0;
    while (cur != NULL)
    {
        result[i] = cur->val + '0';
        cur = cur->next;
        i++;
    }
    result[i] = '\0';
    return strtol(result, NULL, 2);
}

int main()
{
    struct ListNode *head = malloc(sizeof(struct ListNode));
    head->val = 1;
    head->next = malloc(sizeof(struct ListNode));
    head->next->val = 0;
    head->next->next = malloc(sizeof(struct ListNode));
    head->next->next->val = 1;
    head->next->next->next = NULL;

    printf("%d", getDecimalValue(head));
    return 0;
}

File: JavaScript/convert_binary_number_in_a_linked_list_to_integer.js
function ListNode(val, next) {
    this.val = (val === undefined ? 0 : val)
    this.next = (next === undefined ? null : next)
}

/**
 * @param {ListNode} head
 * @return {number}
 */

var getDecimalValue = function (head) {
    let result = ""
    var cur = head

    while (cur !== null) {
        result += cur.val
        cur = cur.next
    }
    return parseInt(result, 2)
};